yt_dlp/utils.py

```diff
@@ -3485,17 +3485,19 @@ def age_restricted(content_limit, age_limit):
     return age_limit < content_limit
 
 
+BOMS = [
+    (b'\xef\xbb\xbf', 'utf-8'),
+    (b'\x00\x00\xfe\xff', 'utf-32-be'),
+    (b'\xff\xfe\x00\x00', 'utf-32-le'),
+    (b'\xff\xfe', 'utf-16-le'),
+    (b'\xfe\xff', 'utf-16-be'),
+]
+""" List of known byte-order-marks (BOM) """
```

Comment:
probly better to be a dict

Reply:
dicts maintain order in cpython3.6. Besides, 3.6 support will be removed in a few days

Reply:
I'm afraid of happening in the future
Also, since the order is important, using dict here is a clear misuse

Reply:
Python 3.7+ dicts are ordered according to specification https://docs.python.org/3.7/library/stdtypes.html#mapping-types-dict
![image](https://user-images.githubusercontent.com/6623388/179210131-4fb9e7e2-b758-47fb-98c0-8df2634af716.png)

Reply:
I do understand that dicts currently maintain its order, and there is one-by-one correspondence. But will it make anything clear?
<!-- Next time I saw the code I'm sure I'm confused and say "Why it's using dict here!" -->